pros/src/async_runtime/mod.rs

```diff
@@ -0,0 +1,51 @@
+use core::{cell::RefCell, future::Future};
+
+use alloc::rc::Rc;
+
+pub(crate) mod executor;
+pub(crate) mod reactor;
+
+pub struct JoinHandle<T> {
+    output: Rc<RefCell<Option<T>>>,
+    _marker: core::marker::PhantomData<T>,
```

Comment:
I don't think the marker field is necessary here.

Reply:
It isn't that should be fixed.

Reply:
done